### PR TITLE
Implemented reusable Checkbox and Label components

### PR DESCRIPTION
**Pull request by frenchykiller**

### CODE REVIEW

**Comment by frenchykiller** [comment]
[As a developer, I want to have an reusable checkbox component.](https://trello.com/c/kaFYJG5k/25-as-a-developer-i-want-to-have-an-reusable-checkbox-component)

File: src/components/shared/form/CheckboxComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+/**
+* @name CheckboxComponent
+* @description A component used to display styled checkbox element.
+*
+* @param {bool}	[disabled]		: Whether the checkbox is disabled.
+* @param {bool}	[checked]		: Whether the checkbox is checked.
+* @param {string}	id				: The id of the checkbox.
+*/
+const CheckboxComponent = ({ id, checked, disabled, ...otherProps }) => (
```
Comment (yannhodiesne) [inline src/components/shared/form/CheckboxComponent.jsx]: A `forwardRef` would be a good idea, to let us choose between having to manage a state (and re-rendering at every click on the checkbox) or accessing the checked state only when needed

**Comment by yannhodiesne** [comment]
`@author` is missing in almost every jsdoc related to this PR

File: src/components/shared/form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { StyledDiv } from 'components/shared/layout';
+import PropTypes from 'prop-types';
+import { tuna } from 'theme/colors';
+import CheckboxComponent from './CheckboxComponent';
+import LabelComponent from './LabelComponent';
+import LegendComponent from './LegendComponent';
+import FieldsetComponent from './FieldsetComponent';
+/**
```
Comment (yannhodiesne) [inline src/components/shared/form/index.jsx]: A blank line is missing

File: src/components/shared/form/index.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @name LabeledCheckbox
+ * @description A component used to display styled checkbox element.
+ *
+ * @param {bool}	[disabled]		: Whether the checkbox is disabled.
+ * @param {bool}	[checked]		: Whether the checkbox is checked.
+ * @param {string}	id				: The id of the checkbox.
+ * @param {string}	children		: text to display in the label.
+ */
+const LabeledCheckbox = ({ children, id, checked, disabled, ...otherProps }) => (
```
Comment (yannhodiesne) [inline src/components/shared/form/index.jsx]: A `forwardRef` would be interesting here too, forwarding it to the internal `CheckboxComponent`

File: src/components/shared/form/index.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	<StyledDiv {...otherProps} display="flex" alignItems="center">
+		<CheckboxComponent disabled={disabled} checked={checked} id={id} />
```
Comment (yannhodiesne) [inline src/components/shared/form/index.jsx]: How do we give it an `onChange` event ?

File: tests/components/shared/form/CheckboxComponent.test.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { shallow } from 'enzyme';
+import CheckboxComponent from 'components/shared/form/CheckboxComponent';
+import { cleanup } from '@testing-library/react';
+
+describe('CheckboxComponent', () => {
```
Comment (yannhodiesne) [inline tests/components/shared/form/CheckboxComponent.test.js:6]: Missing tests for checking/unchecking it and checking if the event is triggered

**Comment by yannhodiesne** [comment]
LabeledComponent needs to be inside its own file, and is missing its unit test suite

File: src/components/shared/form/index.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		<LabelComponent element={id} color={disabled ? tuna.darker1 : 'initial'}>
```
Comment (yannhodiesne) [inline src/components/shared/form/index.jsx:20]: `element={id}` ?

File: tests/components/shared/form/formElements.test.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { shallow } from 'enzyme';
+
+import {
+	LabeledCheckbox,
+} from 'components/shared/form';
+
+describe('Exported form elements', () => {
+	describe('LabeledCheckbox', () => {
```
Comment (yannhodiesne) [inline tests/components/shared/form/formElements.test.js]: Missing tests for checking/unchecking it and checking if the event is triggered

File: src/components/shared/form/index.jsx (hunk 2 above)
Comment (frenchykiller) [reply src/components/shared/form/index.jsx]: fixed

File: src/components/shared/form/index.jsx (hunk 4 above)
Comment (frenchykiller) [reply src/components/shared/form/index.jsx]: `{...otherProps}` moved to `CheckboxComponent`

File: tests/components/shared/form/CheckboxComponent.test.js (hunk 5 above)
Comment (frenchykiller) [reply tests/components/shared/form/CheckboxComponent.test.js:6]: tests added

File: tests/components/shared/form/formElements.test.js (hunk 7 above)
Comment (frenchykiller) [reply tests/components/shared/form/formElements.test.js]: tests added

File: src/components/shared/form/CheckboxComponent.jsx (hunk 1 above)
Comment (frenchykiller) [reply src/components/shared/form/CheckboxComponent.jsx]: added `forwardRef` as requested

File: src/components/shared/form/index.jsx (hunk 3 above)
Comment (frenchykiller) [reply src/components/shared/form/index.jsx]: added `forwardRef` as requested

**Review by yannhodiesne — APPROVED** [review]

**Pull request merged**